### PR TITLE
Allow re-evaluating fuzzy matches against actual message data

### PR DESCRIPTION
**Pull request by ksmurchison**

### CODE REVIEW

File: imap/search_expr.c [hunk 1]
```diff
@@ -2515,7 +2515,7 @@ EXPORTED void search_attr_init(void)
             (void *)message_get_subject
         },{
             "to",
-            SEA_FUZZABLE,
+            SEA_FUZZABLE|SEA_UNFUZZ,
```
Comment (brong) [inline imap/search_expr.c:2518]: Presumably we should be doing "From" here as well.

File: imap/jmap_mail.c [hunk 2]
```diff
@@ -3278,6 +3297,40 @@ static int guidsearch_expr_eval(struct conversations_state *cstate,
...
+                char *mboxname = NULL;
+                msgrecord_t *mr = NULL;
+                message_t *msg = NULL;
+                uint32_t uid;
+                int ret = 0;
+
+                message_guid_decode(&guid, match->guidrep);
+                jmap_set_emailid(&guid, email_id);
+
+                if (!jmap_email_find(e->req, email_id, &mboxname, &uid) &&
```
Comment (brong) [inline imap/jmap_mail.c:3315]: This is a significant extra expense to these queries.

**Review by brong — COMMENTED** [review]
Unfortunately this looks technically correct, but too expensive :(  The VIPs function uses exactly these fields, and requires that the searches are fast.

We're going to have to reindex again.  Hopefully we can do it more cheaply than reindexing everything with some kind of clever "parse the cache record and just index from that".

File: imap/jmap_mail.c (hunk 2 above)
Comment (ksmurchison) [reply imap/jmap_mail.c:3315]: Yes, but I see no way around it with the current indexing.  Maybe limiting this to only when the search string contains '@' but if we're going to re-index then it doesn't matter.

Is there anyway to implement prefix, suffix, and complete string matches via Xapian?  If not, we're stuck doing the above for those matches.

**Pull request closed without merging**